pkg/cmd/server/origin/audit.go

```diff
@@ -0,0 +1,59 @@
+package origin
+
+import (
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/pborman/uuid"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+type auditResponseWriter struct {
+	http.ResponseWriter
+	id string
+}
+
+func (a *auditResponseWriter) WriteHeader(code int) {
+	glog.Infof("AUDIT: %s: response:%d", a.id, code)
+	a.ResponseWriter.WriteHeader(code)
+}
+
+// auditHandler is responsible for logging audit information for all the
+// request coming to server. Each audit log contains two entries:
+// 1. the request line containing:
+//    - unique id allowing to match the response line (see 2)
+//    - source ip of the request
```

Comment:
@pweil- suggested me using `k8s.io/kubernetes/pkg/util/net.GetClientIP` which does exactly that.